app/helpers/automated_tests_helper.rb

```diff
@@ -38,18 +38,25 @@ def update_test_groups_from_specs(assignment, test_specs)
     test_specs_path = assignment.autotest_settings_file
     # create/modify test groups based on the autotest specs
     test_group_ids = []
-    test_specs['testers'].each do |tester_specs|
+    criteria_map = assignment.ta_criteria.pluck(:type, :name, :id).map do |type, name, id_|
+      ["#{type}:#{name}", id_]
+    end.to_h
+    test_specs['testers']&.each do |tester_specs|
       next if tester_specs['test_data'].nil?
 
-      tester_specs['test_data'].each do |test_group_specs|
+      tester_specs['test_data']&.each do |test_group_specs|
```

Comment:
In this case, I think the check on the preceding line ensures that `tester_specs['test_data']` is not nil. I'm hesitant to use `&.` when it's not required.